src/main/kotlin/uk/gov/justice/digital/hmpps/prisonercontactregistry/dto/Address.kt

```diff
@@ -0,0 +1,64 @@
+package uk.gov.justice.digital.hmpps.prisonercontactregistry.dto
+
+import io.swagger.v3.oas.annotations.media.Schema
+import org.springframework.format.annotation.DateTimeFormat
+import java.time.LocalDate
+
+@Schema(description = "An address")
+class Address private constructor(
+  @Schema(description = "Address Type", example = "BUS", required = false) val addressType: String?,
+  @Schema(description = "Flat", example = "3B", required = false) val flat: String?,
+  @Schema(description = "Premise", example = "Liverpool Prison", required = false) val premise: String?,
+  @Schema(description = "Street", example = "Slinn Street", required = false) val street: String?,
+  @Schema(description = "Locality", example = "Brincliffe", required = false) val locality: String?,
+  @Schema(description = "Town/City", example = "Liverpool", required = false) val town: String?,
+  @Schema(description = "Postal Code", example = "LI1 5TH", required = false) val postalCode: String?,
+  @Schema(description = "County", example = "HEREFORD", required = false) val county: String?,
+  @Schema(description = "Country", example = "ENG", required = false) val country: String?,
+  @Schema(description = "Additional Information", example = "This is a comment text", required = false) val comment: String?,
+  @Schema(description = "Primary Address", example = "Y", required = true) val primary: Boolean,
+  @Schema(description = "No Fixed Address", example = "N", required = true) val noFixedAddress: Boolean,
+  @Schema(description = "Date Added", example = "2000-10-31", required = false) val startDate: LocalDate?,
+  @Schema(description = "Date ended", example = "2000-10-31", required = false) val endDate: LocalDate?,
+  @Schema(description = "The phone number associated with the address", required = false) val phones: List<Telephone> = listOf(),
+  @Schema(description = "The address usages/types", required = false) val addressUsages: List<AddressUsage> = listOf()
+) {
+
+  // The person has a list of addresses
```

Comment:
We don't normally use builders in kotlin projects - we use the constructor with specific variable assignment.

It will save a lot of typing in these classes.

Reply:
Thank you for the advice